tests/expose_test.go

```diff
@@ -73,6 +76,35 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 		return job
 	}
 
+	cleanupService := func(serviceName string, namespace string) error {
+		err := virtClient.CoreV1().Services(namespace).Delete(serviceName, &k8smetav1.DeleteOptions{})
+		err = wait.PollImmediate(2*time.Minute, time.Second, func() (done bool, err error) {
+			_, err = virtClient.CoreV1().Services(namespace).Get(serviceName, k8smetav1.GetOptions{})
+			return err != nil && errors.IsNotFound(err), nil
+		})
+		return err
+	}
+
+	cleanupJob := func(jobName string, namespace string) error {
```

Comment:
> * Consider adding to `tests/job.go`

I'd rather not derail this PR further by attempting to define whatever API for job cleanups. I have a simple need in this module, which is addressed by the following code. From [this discussion](https://github.com/kubevirt/kubevirt/pull/4272#discussion_r511586890), it is not even clear we want to cleanup jobs, since, they kind of recycle themselves.

> * I would prefer a function signature that gets a pointer to a job and retrieves from it the name and namespace.

I can compromise, since I see those 2 things as being fundamentally the same. End of the day, I don't need the object to delete, just the name and the namespace. Which I retrieve from the object. Whatever.

> * It may be even nicer to have only one function that accepts a variable number of jobs for deletion.

It is implemented below.

> * Do you really want to use the teardown pattern given what we know?

Given that the test is currently worse than that (resources leak throughout the tests, which mean that when anything gets logged, it logs the resources of more than one test), yes. I think right now what we have is worse.

Reply:
If I need to debug a failing test. I would prefer extra (even non relevant logs) over no logs at all.

Reply:
Stopped removing the resources. We're now back to using the previous approach, where the entities we create get deleted once per context, in the `tests.BeforeAll` .